### PR TITLE
feat: Add wealth of nations chart

### PR DESCRIPTION
**Pull request by mbssantos**
Part of #19

### CODE REVIEW

File: site/src/example-charts/wealth-and-health-of-nations/helpers.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+var helpers = {
+	innerDimensions: function (element) {
```
Comment (mbssantos) [inline site/src/example-charts/wealth-and-health-of-nations/helpers.js]: Has this been moved somewhere?

File: site/src/example-charts/wealth-and-health-of-nations/index.html [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+
+    <title>{{default title 'd3fc'}}</title>
+    <meta name="description" content="A collection of components that make it easy to build interactive charts with D3.">
+
+    <script src="https://d3js.org/d3.v4.js"></script>
```
Comment (mbssantos) [inline site/src/example-charts/wealth-and-health-of-nations/index.html]: TODO: revert

File: site/src/example-charts/wealth-and-health-of-nations/wealth-and-health-of-nations.js [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+
+                overlay.enter()
+                    .on('mouseover', function() { label.classed('active', true); })
+                    .on('mouseout', function() { label.classed('active', false); })
+                    .on('mousemove', mousemove)
+                    .on('touchstart', function() { label.classed('active', true); })
+                    .on('touchend', function() { label.classed('active', false); })
+                    .on('touchmove', mousemove)
+
+                overlay.attr('x', labelBox.x)
```
Comment (mbssantos) [inline site/src/example-charts/wealth-and-health-of-nations/wealth-and-health-of-nations.js]: What's the correct way to position the rect in a d3 world?

**Comment by ColinEberhardt** [comment]
@mbssantos - I'll take a look at your code shortly. However, one thing I am struggling with is running the examples. It is falling over with this error:

```
{ Error: ENOENT: no such file or directory, open '_layouts/undefined.hbs'
    at Error (native)
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '_layouts/undefined.hbs' }
```

I think @alisd23 saw this one before - not sure how to resolve it!

Currently the example charts don't use a pre-existing layout. They should probably use the same `default` layout that the root index page uses.

**Comment by ColinEberhardt** [comment]
Ah - I see an immediate problem here. You've been migrating the much more complex version of this chart:

https://d3fc.io/examples/wealth-and-health-of-nations/index.html

However, the homepage of d3fc uses the much simper version:

https://d3fc.io/examples/bubble/index.html

We are trying to keep things as simple as possible, so my preference is to migrate the simpler one.

Sorry!

File: site/src/example-charts/wealth-and-health-of-nations/helpers.js (hunk 1 above)
Comment (ColinEberhardt) [inline site/src/example-charts/wealth-and-health-of-nations/helpers.js]: This was used previously to measure elements in order to compute a suitable size for SVG elements. This is now entirely internal to d3fc-element.

File: site/src/example-charts/wealth-and-health-of-nations/wealth-and-health-of-nations.js (hunk 3 above)
Comment (ColinEberhardt) [inline site/src/example-charts/wealth-and-health-of-nations/wealth-and-health-of-nations.js]: That's a very open-ended question! With d3fc-element we are favouring using CSS flexbox. This means that for the top-level components of a chart (e.g. axes, labels, series, legend), you'd use separate SVG elements, and flexbox to correctly position them.

**Comment by ColinEberhardt** [comment]
Ah - I see the problem, these example charts aren't actually integrated into the build - you're currently just serving directly from the file-system.

Can you please integrate them with `npm run site`? :-)

**Comment by mbssantos** [comment]
>Ah - I see the problem, these example charts aren't actually integrated into the build - you're currently just serving directly from the file-system

@ColinEberhardt  I've agreed with @alisd23 to create PR's to his branch in this format because we won't have an examples page anymore. Is it alright to have it there until we decide on what it should look like?

> However, the homepage of d3fc uses the much simper version:
https://d3fc.io/examples/bubble/index.html

Well, at least I learned something! Will do it in this PR if that's alright

**Comment by ColinEberhardt** [comment]
> @ColinEberhardt I've agreed with @alisd23 to create PR's to his branch in this format because we won't have an examples page anymore. Is it alright to have it there until we decide on what it should look like?

Ah, no worries, that sounds like a decent plan to get you started.

> Will do it in this PR if that's alright

Sure, happy to see an update to this PR.

**Comment by mbssantos** [comment]
@ColinEberhardt PR updated with wealth of nations bubble chart

File: site/src/example-charts/bubble-chart/index.html [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+
+    <title>{{default title 'd3fc'}}</title>
+    <meta name="description" content="A collection of components that make it easy to build interactive charts with D3.">
+
+    <script src="https://d3js.org/d3.v4.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/d3-legend/2.16.0/d3-legend.min.js"></script>
```
Comment (mbssantos) [inline site/src/example-charts/bubble-chart/index.html]: Is it alright to leave it like this until we know how we're going to include packages?

File: site/src/example-charts/bubble-chart/bubble-chart.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+                )
+        .xDomain(fc.extentLinear()
+          .accessors([d => d.income])(data))
+        .yDomain(fc.extentLinear()
+          .accessors([d => d.lifeExpectancy])(data))
+        .xLabel('Income (dollars)')
+        .yLabel('Life expectancy (years)')
+        .xTicks(2, d3.format(',d'))
+        .chartLabel('The Wealth & Health of Nations')
+        .yOrient('left')
```
Comment (mbssantos) [inline site/src/example-charts/bubble-chart/bubble-chart.js]: This leaves the label pointing the "wrong" way, i.e. an A's tip points right.
Is the right approach to just rotate it -90deg with css or is this something chartSvgCartesian should handle?

Comment (chrisprice) [inline site/src/example-charts/bubble-chart/bubble-chart.js]: See #46 comment

File: site/src/example-charts/bubble-chart/index.html (hunk 4 above)
Comment (chrisprice) [inline site/src/example-charts/bubble-chart/index.html]: Yea

File: site/src/example-charts/bubble-chart/bubble-chart.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+            sel.enter()
+                .attr('fill', function(d) { return color(d.region); });
+        });
+
+    var chart = fc.chartSvgCartesian(
+                  d3.scaleLog(),
+                  d3.scaleLinear()
+                )
+        .xDomain(fc.extentLinear()
+          .accessors([d => d.income])(data))
```
Comment (chrisprice) [inline site/src/example-charts/bubble-chart/bubble-chart.js]: mix of fat-arrow and non-fat-arrow

File: site/src/example-charts/bubble-chart/index.html [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    <script type="text/javascript" src="../../../dist/build/d3fc.js"></script>
+
+    <style>
```
Comment (mbssantos) [inline site/src/example-charts/bubble-chart/index.html]: I'll sort this as part of the build PR I'm doing now if that's alright, most of them will go when they're integrated.

File: site/src/example-charts/bubble-chart/bubble-chart.js (hunk 6 above)
Comment (mbssantos) [inline site/src/example-charts/bubble-chart/bubble-chart.js]: Well spotted - these examples are not being translated so shouldn't use fat arrows at all.
will convert them all to es6 on my current build PR if that's alright?

**Comment by chrisprice** [comment]
:+1:

**Pull request merged**